### PR TITLE
Fix: Do not crash when passing null values to @Nullable methods, eg User and Scope

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Fix: User.setOthers is marked @Nullable but crashes when passing in null

## :bulb: Motivation and Context
#1131

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/Scope.java [hunk 1]
```diff
@@ -175,7 +175,10 @@ public void setUser(@Nullable User user) {
    *
    * @param fingerprint the fingerprint list
    */
-  public void setFingerprint(@NotNull List<String> fingerprint) {
+  public void setFingerprint(final @NotNull List<String> fingerprint) {
+    if (fingerprint == null) {
```
Comment (philipphofmann) [inline sentry/src/main/java/io/sentry/Scope.java:179]: `m`: What about throwing `IllegalArgumentException`?

File: sentry/src/test/java/io/sentry/protocol/UserTest.kt [hunk 2]
```diff
@@ -79,7 +53,28 @@ class UserTest {
...
+
+    @Test
+    fun `setting null others do not crash`() {
+        val user = createUser()
+        user.others = null
+
+        assertNull(user.others)
+    }
+
+    private fun createUser(): User {
```
Comment (philipphofmann) [inline sentry/src/test/java/io/sentry/protocol/UserTest.kt:68]: Nice refactoring 💯

**Review by philipphofmann — COMMENTED** [review]
The changes LGTM, but I think the title of the PR doesn't reflect all the changes in here. We make small fixes in both Scope and User. I think we can either change the title of the PR or we split this up into 2 PRs.

File: sentry/src/main/java/io/sentry/Scope.java (hunk 1 above)
Comment (marandaneto) [reply sentry/src/main/java/io/sentry/Scope.java:179]: we cant do that
https://develop.sentry.dev/sdk/philosophy/#degrade-gracefully
this function is called by the users, not the SDK itself

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@
 * Feat: set isSideLoaded info tags
 * Enhancement: Read tracesSampleRate from AndroidManifest
 * Ref: Return NoOpTransaction instead of null (#1126)
+* Fix: User.setOthers is marked @Nullable but crashes when passing in null #1131
```
Comment (marandaneto) [inline CHANGELOG.md:11]:
```suggestion
* Fix: Do not crash when passing null values to @Nullable methods, eg User and Scope
```

**Comment by marandaneto** [comment]
> The changes LGTM, but I think the title of the PR doesn't reflect all the changes in here. We make small fixes in both Scope and User. I think we can either change the title of the PR or we split this up into 2 PRs.

good catch, title and changelog are renamed

**Review by philipphofmann — APPROVED** [review]
LGTM

**Review by maciejwalkowiak — APPROVED** [review]

**Pull request merged**